be/src/util/counter_cond_variable.hpp

```diff
@@ -60,18 +57,13 @@ class CounterCondVariable {
         _count += inc;
     }
 
-    // decrease the counter, and notify all waiters
+    // decrease the counter, and notify all waiters when counter <= 0
     void dec(int dec = 1) {
         std::unique_lock<std::mutex> lock(_lock);
         _count -= dec;
-        _cv.notify_all();
-    }
-
-    // decrease the counter to zero
-    void dec_to_zero() {
```

Comment:
I understand that the reason you removed the `dec_to_zero()` method is to hope that all related memtables can be sequentially completed through `memtable_flush_queue`. To ensure that there will be no related memtable still in the flush queue before `delta_writer` is closed?

But even after the `delta_writer` is closed, is there any related memtable in the flush queue, does it matter?

Reply:
As I commented in [this thread](https://github.com/apache/incubator-doris/issues/1984#issuecomment-563240199), if `FlushHandle::wait` returns early, `DeltaWriter` could be destructed before its related `MemTable`, which leads to `MemTable` referring to a destructed parent MemTracker.

Reply:
OK